### PR TITLE
Update cellpose defaults & parse additional parameters

### PR DESCRIPTION
**Pull request by jluethi**
Closes https://github.com/fractal-analytics-platform/fractal-tasks-core/issues/314 by providing the full new model zoo as an option, not just the cellpose 1.0 models.

Updating some defaults to be better aligned with our typical data

Also adding a few additional cellpose parameters to be parsed through.

### CODE REVIEW

File: fractal_tasks_core/cellpose_segmentation.py [hunk 1]
```diff
@@ -56,17 +56,37 @@
 
 __OME_NGFF_VERSION__ = fractal_tasks_core.__OME_NGFF_VERSION__
 
+CELLPOSE_MODEL_ZOO = [
```
Comment (tcompa) [inline fractal_tasks_core/cellpose_segmentation.py:59]: What about `from cellpose.models import MODEL_NAMES as CELLPOSE_MODEL_ZOO`?

**Comment by tcompa** [comment]
Since we are already looking at this, it wouldn't hurt to add a few more type hints to `def segment_FOV`

File: fractal_tasks_core/cellpose_segmentation.py [hunk 2]
```diff
@@ -80,6 +100,12 @@ def segment_FOV(
     :param flow_threshold: TBD
     :param label_dtype: TBD
     :param well_id: TBD
+    :param augment: Whether to use cellpose augmentation to tile images
+                    with overlap
+    :param net_avg: Whether to use cellpose net averaging to run the 4 built-in
+                    networks (useful for nuclei, cyto & cyto, not sure it works
```
Comment (tcompa) [inline fractal_tasks_core/cellpose_segmentation.py:106]: typo: "cyto and cyto"

File: fractal_tasks_core/cellpose_segmentation.py [hunk 3]
```diff
@@ -186,6 +216,14 @@ def cellpose_segmentation(
     :param model_type: Parameter of ``CellposeModel`` class.
     :param pretrained_model: Parameter of ``CellposeModel`` class (takes
                              precedence over ``model_type``).
+    :param min_size: Minimum size of the segmented objects (in pixels).
+                     Use -1 to turn off the size filter
+    :param agument: Whether to use cellpose augmentation to tile images
+                    with overlap
+    :param net_avg: Whether to use cellpose net averaging to run the 4 built-in
+                    networks (useful for nuclei, cyto & cyto, not sure it works
```
Comment (tcompa) [inline fractal_tasks_core/cellpose_segmentation.py:224]: Same "cyto & cyto" typo as above

**Comment by tcompa** [comment]
I made some (really minor) comments above, but other than that this PR looks ready.

**Comment by tcompa** [comment]
Re: type hints with cellpose model zoo

It turns out that using `Literal[something]` with a dynamically-generated `something` is not supported (and for good reasons). One way to generate a correct type hint is the one below. Note: if it were "only" for type hints, I'd suggest to ignore this issue. However we also use this type in the Pydantic model for task-arguments validation, and therefore I'd rather have it right.

As an example, this script
```python
from enum import Enum
from cellpose import models
from pydantic.main import BaseModel
from devtools import debug

# Define class based on dynamic list of valid values in models.MODEL_NAMES
ModelFromCellposeZoo = Enum(
    "ModelInCellposeZoo",
    ((value, value) for value in models.MODEL_NAMES),
    type=str,
)

class TaskArguments(BaseModel):
    model: ModelInCellposeZoo
    x: int

# Create a valid TaskArguments instance
valid_args = TaskArguments(model="cyto", x=1)
debug(valid_args)

# Create an invalid TaskArguments instance
invalid_args = TaskArguments(model="mymodel", x=1)
debug(invalid_args)
```
outputs
```
# Valid arguments:

x.py:25 <module>
    valid_args: TaskArguments(
        model=<ModelFromCellposeZoo.cyto: 'cyto'>,
        x=1,
    ) (TaskArguments)

# -----------------------------------------

# Invalid arguments:

Traceback (most recent call last):
  File "/home/tommaso/Fractal/fractal-tasks-core/x.py", line 29, in <module>
    invalid_args = TaskArguments(model="mymodel", x=1)
  File "pydantic/main.py", line 342, in pydantic.main.BaseModel.__init__
pydantic.error_wrappers.ValidationError: 1 validation error for TaskArguments
model
  value is not a valid enumeration member; permitted: 'cyto', 'nuclei', 'tissuenet', 'livecell', 'cyto2', 'general', 'CP', 'CPx', 'TN1', 'TN2', 'TN3', 'LC1', 'LC2', 'LC3', 'LC4' (type=type_error.enum; enum_values=[<ModelFromCellposeZoo.cyto: 'cyto'>, <ModelFromCellposeZoo.nuclei: 'nuclei'>, <ModelFromCellposeZoo.tissuenet: 'tissuenet'>, <ModelFromCellposeZoo.livecell: 'livecell'>, <ModelFromCellposeZoo.cyto2: 'cyto2'>, <ModelFromCellposeZoo.general: 'general'>, <ModelFromCellposeZoo.CP: 'CP'>, <ModelFromCellposeZoo.CPx: 'CPx'>, <ModelFromCellposeZoo.TN1: 'TN1'>, <ModelFromCellposeZoo.TN2: 'TN2'>, <ModelFromCellposeZoo.TN3: 'TN3'>, <ModelFromCellposeZoo.LC1: 'LC1'>, <ModelFromCellposeZoo.LC2: 'LC2'>, <ModelFromCellposeZoo.LC3: 'LC3'>, <ModelFromCellposeZoo.LC4: 'LC4'>])
```

**Comment by tcompa** [comment]
Concerning types: I had not thought about type hints (and task validation) before commenting, but now that Joel stressed how subtle it gets I wouldn't be against a hard-coded list of models [if a new model is introduced (probably this doesn't happen for each patch release..) we'll have to update the list].

TL;DR I'm fine either ways.

**Comment by jluethi** [comment]
Thanks a lot for the options shown here @tcompa . I went with the first one to use an Enum and added a few more docstrings & type hints for `segment_FOV`.

Should be ready to merge once the tests pass

**Comment by tcompa** [comment]
> Thanks a lot for the options shown here @tcompa . I went with the first one to use an Enum and added a few more docstrings & type hints for `segment_FOV`.
> 
> Should be ready to merge once the tests pass

Perfect, thanks. I just added some `Optional` (for arguments that can also be `None`), and this is good to merge.

**Comment by jluethi** [comment]
Awesome, thanks @tcompa
